File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/fragmentlifecycle/viewpager/PagerAdapter.java
package net.grandcentrix.thirtyinch.sample.fragmentlifecycle.viewpager;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import net.grandcentrix.thirtyinch.sample.fragmentlifecycle.TestFragmentA;

class PagerAdapter extends FragmentPagerAdapter {

    public PagerAdapter(final FragmentManager fm) {
        super(fm);
    }

    @Override
    public int getCount() {
        return 4;
    }

    @Override
    public Fragment getItem(final int position) {
        return new TestFragmentA();
    }
}

File: sample/src/main/java/net/grandcentrix/thirtyinch/sample/fragmentlifecycle/viewpager/PagerAdapter.kt
package net.grandcentrix.thirtyinch.sample.fragmentlifecycle.viewpager

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import net.grandcentrix.thirtyinch.sample.fragmentlifecycle.TestFragmentA

internal class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getCount(): Int {
        return 4
    }

    override fun getItem(position: Int): Fragment {
        return TestFragmentA()
    }
}
